src/mlpack/methods/ann/layer/isrlu.hpp

```diff
@@ -0,0 +1,169 @@
+/**
+ * @file isrlu.hpp
+ * @author Pranav Reddy P
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_ISRLU_HPP
+#define MLPACK_METHODS_ANN_LAYER_ISRLU_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The ISRLU function, defined by
+ * 
+ * @f{eqnarray*}{
+ * f(x) &=& x/sqrt(1 + alpha*x*x) if x<0
+ *      &=& x                     if x>=0
+ *
+ * f'(x) &=& cube(1/sqrt(1 + alpha*x*x)) if x<0
+ *       &=& 1                           if x>=0
+ * @f}
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class ISRLU
+{
+ public:
+  /**
+   * Create the ISRLU object using the specified parameters.
+   * The non zero gradient can be adjusted by specifying the parameter
+   * alpha in the range 0 to 1. Default (alpha = 0.03)
```

Comment:
Also if alpha can be negative write it as -1 to 1  or magnitude of alpha between 0 and 1.

Reply:
Hey @kartikdutt18 This is something I seemed to have missed. Alpha can be negative but this could lead to an error if x < -1/sqrt(alpha) since we would then be trying to find square root of a negative number. How do you think I should define the range of alpha over here?

Reply:
I think domain of x would be - 1 / sqrt(alpha) to 1 (or infinity)

Reply:
Could you add a link to the documentation if you have any. I would also find something to clear this up. Thanks.

Reply:
Yes that's the domain of x. Should I specify that also somewhere? And what should the range of alpha be specified as?

Reply:
Add a fatal error for that. Use `Log:: Fatal <<" Error message goes here."<< std::endl;`